src/sentry/notifications/notifications/base.py

```diff
@@ -1,18 +1,21 @@
+import abc
 from typing import TYPE_CHECKING, Any, Mapping, MutableMapping, Optional, Tuple, Union
 
+from sentry import analytics
+from sentry.types.integrations import ExternalProviders
 from sentry.utils.http import absolute_uri
 
 if TYPE_CHECKING:
-    from sentry.models import Project, Team, User
+    from sentry.models import Organization, Project, Team, User
 
 
-class BaseNotification:
+class BaseNotification(abc.ABC):
     fine_tuning_key: Optional[str] = None
     is_message_issue_unfurl = False
+    analytics_event = None
 
-    def __init__(self, project: "Project") -> None:
-        self.project = project
-        self.organization = self.project.organization
+    def __init__(self, organization: "Organization") -> None:
```

Comment:
@mgaeta IMO none of the base notification classes really need a constructor. Let the subclasses define the constructor method. Easy to just add a class property to access the required fields (eg: `organization`) if you just have a project rather than storing the organization as a separate field.